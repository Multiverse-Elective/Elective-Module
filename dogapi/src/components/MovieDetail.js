// src/components/MovieDetail.js
import React from 'react';

const MovieDetail = ({ movie }) => {
  if (!movie) {
    return null;
  }

  const {
    Title,
    Released,
    Year,
    Rated,
    Runtime,
    Genre,
    Director,
    Writer,
    Actors,
    Plot,
    Poster,
  } = movie;

  return (
    <div>
      <h2>{Title}</h2>
      <img src={Poster} alt={Title} />
      <p><strong>Released:</strong> {Released}</p>
      <p><strong>Year:</strong> {Year}</p>
      <p><strong>Rated:</strong> {Rated}</p>
      <p><strong>Runtime:</strong> {Runtime}</p>
      <p><strong>Genre:</strong> {Genre}</p>
      <p><strong>Director:</strong> {Director}</p>
      <p><strong>Writer:</strong> {Writer}</p>
      <p><strong>Actors:</strong> {Actors}</p>
      <p><strong>Plot:</strong> {Plot}</p>
    </div>
  );
};

export default MovieDetail;
