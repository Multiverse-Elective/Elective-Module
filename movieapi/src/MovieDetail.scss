// MovieDetail.scss

.movie-detail {
    display: flex;
    flex-direction: column;
    align-items: flex-start; /* Align items to the left */
    margin: 20px;
  
    img {
      width: 20%;
      height: auto;
      margin-bottom: 10px;
    }
  
    .movie-title {
      font-weight: bold;
      font-size: 20px; /* Adjust the font size as needed */
    }
  
    p {
      margin-bottom: 5px;
    }
  
  }
  