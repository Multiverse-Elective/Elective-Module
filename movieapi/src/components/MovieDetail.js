import React from 'react';
import '../MovieDetail.scss';
const MovieDetail = ({ movie, handleBackButtonClick }) => {
  if (!movie) {
    return null;
  }

  const {
    Title,
    Released,
    Year,
    Rated,
    Runtime,
    Genre,
    Director,
    Writer,
    Actors,
    Plot,
    Poster,
  } = movie;

  return (
    <div className="movie-detail">
      <img src={Poster} alt={Title} />
      <p className="movie-title">{Title}</p>
      <p>Released: {Released}</p>
      <p>Year: {Year}</p>
      <p>Rated: {Rated}</p>
      <p>Runtime: {Runtime}</p>
      <p>Genre: {Genre}</p>
      <p>Director: {Director}</p>
      <p>Writer: {Writer}</p>
      <p>Actors: {Actors}</p>
      <p>Plot: {Plot}</p>
    </div>
  );
};

export default MovieDetail;