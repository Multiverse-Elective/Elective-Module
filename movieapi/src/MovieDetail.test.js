import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import MovieDetail from './components/MovieDetail';

describe('MovieDetail Component', () => {
    const mockMovie = {
      Title: 'Inception',
      Released: '2010-07-16',
      Year: '2010',
      Rated: 'PG-13',
      Runtime: '148 min',
      Genre: 'Action, Adventure, Sci-Fi',
      Director: 'Christopher Nolan',
      Writer: 'Christopher Nolan',
      Actors: 'Leonardo DiCaprio, Joseph Gordon-Levitt, Ellen Page',
      Plot: 'A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a C.E.O.',
      Poster: 'inception-poster.jpg',
    };
  
    it('displays the correct release date', () => {
      render(<MovieDetail movie={mockMovie} />);
      const releaseDateElement = screen.getByText(/Released: \d{4}-\d{2}-\d{2}/);
      expect(releaseDateElement).toBeInTheDocument();
    });
  
    it('displays the correct genre', () => {
      render(<MovieDetail movie={mockMovie} />);
      const genreElement = screen.getByText(/Genre: Action, Adventure, Sci-Fi/);
      expect(genreElement).toBeInTheDocument();
    });
  
    it('displays the correct director', () => {
      render(<MovieDetail movie={mockMovie} />);
      const directorElement = screen.getByText(/Director: Christopher Nolan/);
      expect(directorElement).toBeInTheDocument();
    });
  
    it('displays the correct actors', () => {
      render(<MovieDetail movie={mockMovie} />);
      const actorsElement = screen.getByText(/Actors: Leonardo DiCaprio, Joseph Gordon-Levitt, Ellen Page/);
      expect(actorsElement).toBeInTheDocument();
    });
  
    it('displays the correct plot', () => {
      render(<MovieDetail movie={mockMovie} />);
      const plotElement = screen.getByText(/A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a C.E.O./);
      expect(plotElement).toBeInTheDocument();
    });
  
    it('displays the correct poster', () => {
      render(<MovieDetail movie={mockMovie} />);
      const posterElement = screen.getByAltText('Inception');
      expect(posterElement).toBeInTheDocument();
    });
  
    it('matches snapshot', () => {
      const { asFragment } = render(<MovieDetail movie={mockMovie} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  