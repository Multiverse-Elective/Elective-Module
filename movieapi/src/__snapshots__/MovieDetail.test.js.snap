// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie-detail"
  >
    <img
      alt="Inception"
      src="inception-poster.jpg"
    />
    <p
      class="movie-title"
    >
      Inception
    </p>
    <p>
      Released: 2010-07-16
    </p>
    <p>
      Year: 2010
    </p>
    <p>
      Rated: PG-13
    </p>
    <p>
      Runtime: 148 min
    </p>
    <p>
      Genre: Action, Adventure, Sci-Fi
    </p>
    <p>
      Director: Christopher Nolan
    </p>
    <p>
      Writer: Christopher Nolan
    </p>
    <p>
      Actors: Leonardo DiCaprio, Joseph Gordon-Levitt, Ellen Page
    </p>
    <p>
      Plot: A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a C.E.O.
    </p>
  </div>
</DocumentFragment>
`;
